ClientFactory/DummyFactory.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Http\HttplugBundle\ClientFactory;
+
+use Http\Client\HttpClient;
+
+/**
+ * This class is used with the dependency injection when we register the service. It acts like a place holder and
+ * is needed for the framework to be able to bind method class to it.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class DummyFactory implements ClientFactoryInterface
+{
+    /**
+     * Configure and return a client.
+     *
+     * @param array $config
+     *
+     * @return HttpClient
+     */
+    public function configure(array $config = array())
+    {
+        return;
```

Comment:
i would prefer to throw an exception explaining that something went wrong. null is always a pain to debug.